opsdroid/connector/slack/connector.py

```diff
@@ -205,14 +205,23 @@ async def _get_channels(self):
             cursor = None
 
             while True:
-                channels = await self.slack_web_client.conversations_list(
-                    cursor=cursor, limit=self.channel_limit
-                )
-                self.known_channels.update({c["name"]: c for c in channels["channels"]})
-                cursor = channels["response_metadata"].get("next_cursor")
-
-                if not cursor:
-                    break
+                try:
+                    channels = await self.slack_web_client.conversations_list(
+                        cursor=cursor, limit=self.channel_limit
+                    )
+                    self.known_channels.update({c["name"]: c for c in channels["channels"]})
+                    cursor = channels["response_metadata"].get("next_cursor")
+
+                    if not cursor:
+                        break
+                except SlackApiError as error:
+                    if "ratelimited" in str(error):
+                        wait_time = float(error.response.headers["Retry-After"])
```

Comment:
when the exception is "ralimited" error, there is always a Retry-After header to tell us how long we should wait. Yes, it seems slack API always sends back 30s but the idea is to always respect such header? The other alternative would be to check if there is "Retry-After" in the header first, and if so, honour its value. Otherwise, use a hard-coded value of 30s? What do you think?

Reply:
Yeah I think being defensive here is a good idea. I think we can do something like:

```suggestion
                        wait_time = float(error.response.headers.get("Retry-After", 30)
```